programs/pluto/ipsec_doi.c

```diff
@@ -463,24 +463,20 @@ ipsecdoi_replace(struct state *st
 		 , lset_t policy_add, lset_t policy_del
 		 , unsigned long try)
 {
-	initiator_function *initiator;
+    initiator_function *initiator;
+    struct connection *c = st->st_connection;
     int whack_sock = dup_any(st->st_whack_sock);
     lset_t policy = st->st_policy;
     so_serial_t  newstateno;
 
-    struct state *old_parent_state = state_with_serialno(st->st_clonedfrom);
```

Comment:
I remember adding this, and it made sense at the time.
Hmm.

Reply:
It *does* make sense, but it fails to grab a later parent SA if there is one available.